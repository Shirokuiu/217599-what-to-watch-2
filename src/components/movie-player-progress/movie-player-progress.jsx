import React from "react";
import PropTypes from "prop-types";

export const MoviePlayerProgress = (props) => {
  const {position, lineProgressRef, playPauseRef, duration} = props;
  const moveStyles = {
    left: `${position}%`
  };

  return (
    <div className="player__controls-row">
      <div className="player__time">
        <progress
          className="player__progress"
          ref={lineProgressRef}
          value={position}
          max="100"
        />
        <div
          className="player__toggler"
          ref={playPauseRef}
          style={moveStyles}
        >
          Toggler</div>
      </div>
      <div className="player__time-value">{duration}</div>
    </div>
  );
};

MoviePlayerProgress.propTypes = {
  position: PropTypes.number.isRequired,
  duration: PropTypes.string.isRequired,
  lineProgressRef: PropTypes.shape({
    current: PropTypes.instanceOf(Element)
  }).isRequired,
  playPauseRef: PropTypes.shape({
    current: PropTypes.instanceOf(Element)
  }).isRequired
};
