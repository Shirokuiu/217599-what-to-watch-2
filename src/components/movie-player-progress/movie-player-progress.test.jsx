import React from "react";
import renderer from "react-test-renderer";

import {MoviePlayerProgress} from "./movie-player-progress";

it(`Is movie player progress rendered`, () => {
  const tree = renderer.create(<MoviePlayerProgress
    position={10}
    duration={`0:12:12`}
    lineProgressRef={React.createRef()}
    playPauseRef={React.createRef()}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
