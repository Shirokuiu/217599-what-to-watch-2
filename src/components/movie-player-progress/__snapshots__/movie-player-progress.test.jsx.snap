// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is movie player progress rendered 1`] = `
<div
  className="player__controls-row"
>
  <div
    className="player__time"
  >
    <progress
      className="player__progress"
      max="100"
      value={10}
    />
    <div
      className="player__toggler"
      style={
        Object {
          "left": "10%",
        }
      }
    >
      Toggler
    </div>
  </div>
  <div
    className="player__time-value"
  >
    0:12:12
  </div>
</div>
`;
